apps/test/unit/javalab/JavalabCaptchaDialogTest.js

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import {expect} from '../../util/reconfiguredChai';
+import sinon from 'sinon';
+import {shallow} from 'enzyme';
+import javalabMsg from '@cdo/javalab/locale';
+import {UnconnectedJavalabCaptchaDialog as JavalabCaptchaDialog} from '@cdo/apps/javalab/JavalabCaptchaDialog';
+import ReCaptchaDialog from '@cdo/apps/templates/ReCaptchaDialog';
+
+describe('JavalabCaptchaDialog', () => {
+  let defaultProps,
+    appendNewlineToConsoleLogSpy,
+    appendOutputLogSpy,
+    setDialogOpenSpy,
+    fetchSpy;
+
+  beforeEach(() => {
+    fetchSpy = sinon.stub(window, 'fetch');
```

Comment:
Wasted a lot of time before I realized you can't stub fetch calls using sinon's fakeServer...not sure if there's a better way to do this, but this is what I saw in a few tests that have done something similar.

Reply:
This seems fine to me? You're trying to test the functionality of the onSubmit function, and I don't know if there's any better way than just grabbing it from the props of the child component.

Reply:
Yeah I guess I've heard something like "do the UI action (eg, click on something) and then assert on behavior", but not really possible in this case unless I want to mount the whole component. Since this component is really just a wrapper, maybe it makes sense that I'm asserting directly on the prop.